core/src/main/scala/kafka/server/metadata/BrokerMetadataPublisher.scala

```diff
@@ -80,7 +80,7 @@ object BrokerMetadataPublisher extends Logging {
       Option(newTopicsImage.getPartition(topicId, partitionId)) match {
         case Some(partition) =>
           if (!partition.replicas.contains(brokerId)) {
-            info(s"Found stray log dir $log: the current replica assignment ${partition.replicas} " +
+            info(s"Found stray log dir $log: the current replica assignment ${partition.replicas.mkString("Array(", ", ", ")")} " +
```

Comment:
This actually should have printed the memory reference and not the contents. With the current version, it prints the contents.